src/main/kotlin/no/nav/syfo/infrastructure/database/repository/VurderingRepository.kt

```diff
@@ -51,6 +51,15 @@ class VurderingRepository(private val database: DatabaseInterface) : IVurderingR
                     pdf = pdf,
                 )
             }
+            if (vurdering.isForhandsvarsel()) {
+                if (vurdering.varsel == null)
+                    throw IllegalStateException("Vurdering should have Varsel when creating forhåndsvarsel")
+                connection.createVarsel(
+                    vurderingId = pVurdering.id,
+                    varsel = vurdering.varsel,
+                )
+            }
+
```

Comment:
Dette kunne vi egentlig enforcet med noe `sealed class`'es slik at alle vurderinger som er forhåndsvarsel har `Varsel` 🤔 Ikke noe du trenger å gjøre nå. Bare en tanke 😊

Reply:
Så hadde klassestrukturen alltid sørget for at et `Forhandsvarsel` alltid har et `Varsel` 😊
```kotlin
sealed interface VurderingType {
    class Forhandsvarsel(varsel: Varsel) : VurderingType
    class Oppfylt() : VurderingType
    class Avslag() : VurderingType
}
```

Reply:
Jepp, jeg og Anders snakket litt om noe sånt tidligere, så det hadde vært kult å få til 🙌🏼 Men ja, høres litt ut som en egen refaktorering.